cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/MockCruiseControl.java

```diff
@@ -0,0 +1,158 @@
+                                .withHeaders(header("User-Task-ID", "cruise-control-state")
+                                ));
+
+        // Verbose response
+        String jsonVerbose = getJsonFromResource("CC-State-verbose.json");
+
+        ccServer
+                .when(
+                        request()
+                                .withMethod("GET")
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.JSON.key, "true"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.VERBOSE.key, "true"))
+                                .withPath(CruiseControlEndpoints.STATE.path))
+                .respond(
+                        response()
+                                .withBody(jsonVerbose)
+                                .withHeaders(header("User-Task-ID", "cruise-control-state-verbose")
+                                ));
+
+    }
+
+    private static void setupCCRebalanceResponse(ClientAndServer ccServer) throws IOException, URISyntaxException {
+
+        // Rebalance response with no goals set - non-verbose
+        String json = getJsonFromResource("CC-Rebalance-no-goals.json");
+
+        ccServer
+                .when(
+                        request()
+                                .withMethod("POST")
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.JSON.key, "true"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.DRY_RUN.key, "true|false"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.VERBOSE.key, "false"))
+                                .withPath(CruiseControlEndpoints.REBALANCE.path))
+                .respond(
+                        response()
+                                .withBody(json)
+                                .withHeaders(header("User-Task-ID", "rebalance-no-goals")));
+
+        // Rebalance response with no goals set - verbose
+        String jsonVerbose = getJsonFromResource("CC-Rebalance-no-goals-verbose.json");
+
+        ccServer
+                .when(
+                        request()
+                                .withMethod("POST")
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.JSON.key, "true"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.DRY_RUN.key, "true|false"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.VERBOSE.key, "true"))
+                                .withPath(CruiseControlEndpoints.REBALANCE.path))
+                .respond(
+                        response()
+                                .withBody(jsonVerbose)
+                                .withHeaders(header("User-Task-ID", "rebalance-no-goals-verbose")));
+
+    }
```

Comment:
Are we going to use this mocked server even for tests which involve the operator? I would say yes but at this point we should need a mocked server which provides two options:
1. providing the optimization proposal just on the first request from the client
2. not providing the optimization proposal from the client so that we can test the operator "polling" the user task status and getting the final optimization result after X times of polling (this is kind of what my mocked client do).
Wdyt?

Reply:
Yes, the intention is to use this mocked server in the operator + client integration tests. The mockserver library lets you set a number of attempts before a response or a timespan after first request before response is sent.

I was thinking we could use the "User Task ID" the client sends as a way to tell the mockserver what kind of behaviour we want it to use.

Reply:
Remember that not just the user tasks request could return you that the optimization result is not ready yet but even the initial rebalance proposal request and in that case, you don't have a user task id yet.

Reply:
Yes I will add the async option to those mock end points too